src/backend/cdb/motion/ic_udpifc.c

```diff
@@ -1788,7 +1788,7 @@ destroyConnHashTable(ConnHashTable *ht)
 static inline void
 sendControlMessage(icpkthdr *pkt, int fd, struct sockaddr *addr, socklen_t peerLen)
 {
-	int			n;
+	int			n = 0;
```

Comment:
This is redundant initialization. I suggest to delete this line and rewrite the loop at the end of the function:
```c
	/* retry 10 times for sending control message */
	for (int counter = 0; counter < 10; counter++)
	{
		int n = sendto(fd, (const char *) pkt, pkt->len, 0, addr, peerLen);
		if (n < 0)
		{
			if (errno == EINTR || errno == EAGAIN || errno == EWOULDBLOCK)
				continue;
			else {
				write_log("sendcontrolmessage: got errno %d", errno);
				return;
			}
		}

		if (n < pkt->len)
			write_log("sendcontrolmessage: got error %d errno %d seq %d", n, errno, pkt->seq);
		break;
	}
```